### ISSUE TITLE
Check the type of error instances in tests

### ISSUE THREAD

**Issue by szmarczak** [sindresorhus/got#1087]

**Comment by goto1** [comment]
Hi, could you give an example of what you're looking for?

**Comment by szmarczak** [comment]
```
t.throws(..., {
    message: '...',
    instanceOf: ErrorClassHere
});
```

**Comment by goto1** [comment]
Got it. I can take a look at this.

**Comment by goto1** [comment]
@szmarczak if I change this - https://github.com/sindresorhus/got/blob/master/test/arguments.ts#L18 -  to a different type of error, for example `URIError`, the test should fail when I run `npm test`, correct? For some reason I am having a hard time failing tests with the `npm test` command. Any instructions would be appreciated. 

Also, I see that some tests have `@ts-ignore Error tests` added as a comment and some don't - in what cases this should be added as a comment?

Thanks.

**Comment by szmarczak** [comment]
> if I change this - https://github.com/sindresorhus/got/blob/master/test/arguments.ts#L18 - to a different type of error, for example URIError, the test should fail when I run npm test, correct?

That's right.

> Also, I see that some tests have @ts-ignore Error tests 

The `@ts-ignore` comment is meant to ignore TypeScript errors. E.g. we use that when it's needed to check whether it rejects when passing a value of not compliant type.

> For some reason I am having a hard time failing tests with the npm test command. Any instructions would be appreciated.

Give me a link to your code please, I'll drop you some hints.

**Comment by goto1** [comment]
@szmarczak so I change this line of code from `instanceOf: TypeError` to `instanceOf: URIError` and I still get the same amount of passed and failed tests:

- https://github.com/goto1/got/blob/test-example/test/arguments.ts#L18

Here's what I am seeing before changing that line (`instanceOf: TypeError`):

- https://imgur.com/7tubCon

And here's what I am seeing after (`instanceOf: URIError`, also ran `npx ava reset-cache` just in case):

- https://imgur.com/83EtTAe

Here's the repo:

- https://github.com/goto1/got/tree/test-example

This is my first time working with `ava` so I am going through the docs and trying to get familiar with it. Any tips would be really appreciated. 

Thanks.

**Comment by szmarczak** [comment]
I'll check on this in a few hours.

**Comment by szmarczak** [comment]
Sorry for late response. Regarding the DNS error, it should be fixed in #1051 I'll get back to you once it's merged.
